src/tribler/core/components/libtorrent/download_manager/download_manager.py

```diff
@@ -362,17 +362,24 @@ def get_session(self, hops=0):
 
         return self.ltsessions[hops]
 
-    def set_proxy_settings(self, ltsession, ptype, server=None, auth=None):
+    def set_proxy_settings(self, ltsession, ptype, server: Optional[Tuple] = None, auth: Optional[List] = None):
         """
         Apply the proxy settings to a libtorrent session. This mechanism changed significantly in libtorrent 1.1.0.
         """
+        proxy_port = None
+
+        try:
+            proxy_port = int(server[1])
```

Comment:
Since `[server](server: Optional[Tuple] = None)`, this can raise `TypeError: 'NoneType' object is not subscriptable`

Reply:
Thank you! You are right. 
It has been fixed.